### ISSUE TITLE
METAL GEAR SOLID 2: Sons of Liberty - Master Collection Version (2131640)

### ISSUE THREAD

**Issue by Franatix93** [ValveSoftware/Proton#7394]
# Compatibility Report
- Name of the game with compatibility issues: Metal Gear Solid 2
- Steam AppID of the game: 2131640

## System Information
- Steam Deck
- Proton Hot Fix, Experimental 

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-2131640.log](https://github.com/ValveSoftware/Proton/files/14011639/steam-2131640.log)

## Symptoms 

There is currently a Proton issue users are experiencing in which all the RAM will be filled up and cause the game to crash. I have found the issue happened on a handful of FMV's, as skipping certain videos will stop the bug from happening. All of the cutscenes/FMV's between Vamp Boss Battle (Shell 2 Core) and escorting EE across the fence seem to have this game crashing bug. Memory leaks also happen during gameplay in this specific area 'Shell 2 Core B1' The leak stops when you leave this area. I have attached a video below:
https://www.youtube.com/watch?v=dzGjbQEwRPA&t=4s

Please look into this
Thank you

--

## Reproduction

Cutscenes during Vamp Boss Battle (Shell 2 Core) and escorting EE across the fence seem to have this bug.

**Comment by Franatix93** [comment]
https://youtu.be/dzGjbQEwRPA?si=vHjc9dnQqkNxNWmS

Memory Leak exclusively in this area of the game. RAM will fill until it crashes

---

METAL GEAR SOLID 2: Sons of Liberty - Master Collection Version (2131640)

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7431.
@Franatix93 posted on 2024-01-22T14:44:15:

# Compatibility Report
- Name of the game with compatibility issues: Metal Gear Solid 2: Sons of Liberty - Master Collection Version
- Steam AppID of the game: 2131640

## System Information
Steam Deck
- Proton version: Hot Fix

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-2131640.log](https://github.com/ValveSoftware/Proton/files/14011526/steam-2131640.log)



## Symptoms
Memory leaks happen during gameplay in this specific area 'Shell 2 Core B1' The leak stops when you leave this area. I have attached a video below:
https://www.youtube.com/watch?v=dzGjbQEwRPA&t=4s

## Reproduction
Can occur randomly in this stage of the game

**Comment by Franatix93** [comment]
Tried with latest Proton 8.0-5

Memory leak still happens in this area

**Comment by stephensrmmartin** [comment]
In addition, in the compatdata Temp directory, there are > 32k files, totaling to over 30GB of ".TMP" files, which are playable in VLC. 
My game crashed soon after the area mentioned here, and I think it was due to the FMV sequence in a codec call soon after. My machine was using tons of memory at the time of the crash.

**Comment by Franatix93** [comment]
Yeah I've had to occasionally delete the contents of the TMP folder

**Comment by Franatix93** [comment]
Just ran the new game update (1.4.1) on my OLED Deck. There was a swimming section you could always trigger a memory leak on, and that has been fixed by the looks of things! Tried a few times to trigger it but RAM holds steady

Did noticed the occasional FPS dips in this area (Shell 2 Core B1) but they are really mild

Codec videos have also been fixed. No more green/blurry artefacts

In game monitors (Tanker projector, Uploading EE’s virus etc) have also been fixed

**Comment by Franatix93** [comment]
The game pulls almost locked 3500MHz CPU whilst the GPU is basically idle. I have used PowerTools to adjust frequencies for both in real time to understand what’s happening. But you can cap the CPU to around 1700MHz and still maintain a pretty consistent 60fps. The GPU will still idle at around 400MHz, and any frame rate drops will disappear completely if you cap the minimum GPU frequency to around 1000MHz

It’s as if the .exe doesn’t want to utilise the dedicated GPU

**Comment by kisak-valve** [comment]
METAL GEAR SOLID 2: Sons of Liberty - Master Collection Version

Issue transferred from https://github.com/ValveSoftware/Proton/issues/9943.
@tomcatmew posted on 2026-07-03T12:03:27:

# Compatibility Report
- Name of the game with compatibility issues: METAL GEAR SOLID 2: Sons of Liberty - Master Collection Version
- Steam AppID of the game: 2131640

## System Information
- Steam Machine
- Proton version: 12.0 

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.



Sorry I don't know how to retrieve the log from steam machine. 


## Symptoms 
The game's vocal audio wasn't played at all both in cutscene and in-game. 
After entering game, you can easily find out that the NPC vocal audio is not playing at all.
But the sound effect that such as rainy sound and gun firing sounds are playing correct. 

## Reproduction
Start the game, choose English as language, start First mission .

**Comment by Jayman2000** [comment]
> The game's vocal audio wasn't played at all both in cutscene and in-game.
> After entering game, you can easily find out that the NPC vocal audio is not playing at all.
> But the sound effect that such as rainy sound and gun firing sounds are playing correct.

I can reproduce that same issue. Here’s a full report from my machine:

---

## System Information
- GPU: AMD Radeon RX 580 Series
- Video driver version: Mesa 26.1.5 
- Kernel version: 6.18.42
- Link to full system information report as [Gist](https://gist.github.com/): <https://gist.github.com/Jayman2000/fb262702f53e9034950997b01065d94f>
- Proton version: 1785137693 experimental-11.0-20260724c

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game. (I did find an existing bug report, and I’m adding to it).
- [x] that I have checked whether there are updates for my system available.



[steam-2131640.log](https://github.com/user-attachments/files/30749643/steam-2131640.log)

## Symptoms 

When running MGS2MC natively on Windows 11, audio will play during the intro cutscene. When running MGS2MC via Proton, the audio for the intro cutscene (and many others) will be missing.

## Reproduction

1. Launch METAL GEAR SOLID 2: Sons of Liberty - Master Collection Version.

1. Wait for the Konami logo and the content warning messages to go away.

1. If prompted, agree to the Terms of Use.

1. At this point, you should be at a screen that says “PRESS ANY BUTTON”. Press the <kbd>Enter</kbd> key.

1. At this point, you should be at a menu where you have to choose from a list of menu items. Choose the “Game Selection” menu item.

1. At this point, you should be at a “Language Selection” screen. Choose “English”.

1. At this point, you should be at a screen that’s titled “Start Menu”. Choose “Start Game”.

1. Wait for the Konami, Kojima Productions and Bluepoint Games logos to appear and then disappear.

1. At this point, the intro cutscene should start playing. Watch the intro cutscene, and pay attention to whether or not you hear any audio.

## Workaround

You can work around this problem by installing [the MGS2MC Better Audio Mod](https://www.nexusmods.com/metalgearsolid2mc/mods/3?tab=description). Specifically, you should install these two files that the mod provides:

- “MGS2MC Better Audio Mod 2.0 - Full Version”

- “HQ Audio For Ending Cutscenes - NOT RECOMMENDED”

> [!WARNING]  
> According to the documentation for the MGS2MC Better Audio Mod, the HQ Audio For Ending Cutscenes is “NOT RECOMMENDED”. You should install it anyway, even though it is not recommended. [If you do not install it, then the audio for part of the game’s ending and all of the game’s credits will be missing.](https://github.com/ShizCalev/MGS2-Community-Bugfix-Compilation/issues/11#issuecomment-5134953485)

**Comment by Franatix93** [comment]
Proton can fix this issue simply by adding `xaudio2_9.dll`

Work around until this is fixed is simply paste `xaudio2_9.dll` to your game directory (where the .exe is and add `WINDLLOVERRIDES="xaudio2_9=n" %command%` to your launch options
